### PR TITLE
Kwxm/spec/no flat for bls

### PR DESCRIPTION
**Pull request by kwxm**
This updates the specifcation to reflect the changes made in #5663, where we removed direct serialisation of the BLS12-31 G1 and G2 element types.  The main changes are in Note 4 on pages 38-39 and Section C.3.4 on pages 55-56.  There's a bit of repetition, but I thought that would probably make it more obvious what was going on.  I'll follow this up shortly with a separate PR to add the concrete syntax for lists/pairs/data.

Rendered version: [plutus-core-specification.pdf](https://github.com/IntersectMBO/plutus/files/13716850/plutus-core-specification.pdf)

### CODE REVIEW

File: doc/plutus-core-spec/cardano/builtins1.tex
Comment (kwxm) [inline doc/plutus-core-spec/cardano/builtins1.tex:1]: I added these changes for the list/pair/data issue but then decided to do that in a separate PR.  They'll make more sense when I do that.

File: doc/plutus-core-spec/notation.tex [hunk 1]
```diff
@@ -17,7 +17,7 @@ \subsection{Sets}
   \item $\B^*$ is the set of all bytestrings.
   \item $\Z = \{\ldots, -2, -1, 0, 1, 2, \ldots\}$.
   \item $\mathbb{F}_q$ denotes a finite field with $q$ elements ($q$ a prime power).
-  \item $\mathbb{F}_q^*$ denotes the multiplicative group of nonzero elements of $\mathbb{F}_q$.
+  \item $\units{\mathbb{F}} = \mathbb{F}\setminus\{0\}$ denotes the multiplicative group of nonzero elements of a field $\mathbb{F}$.
```
Comment (kwxm) [inline doc/plutus-core-spec/notation.tex:20]: This is to avoid confusion with the X^* notation for sequences of elements of a set X.  Both F^* and F^× are used in the mathematical literature.

**Comment by kwxm** [comment]
I'll just merge this: it's not very complicated and the reviewers are on holiday.

**Review by mjaskelioff — COMMENTED** [review]
LGTM
